linera-views/src/register_view.rs

```diff
@@ -58,17 +60,26 @@ where
     }
 
     fn rollback(&mut self) {
+        self.delete_storage_first = false;
         self.update = None;
         *self.hash.get_mut() = self.stored_hash;
     }
 
     fn flush(&mut self, batch: &mut Batch) -> Result<(), ViewError> {
-        if let Some(value) = self.update.take() {
+        if self.delete_storage_first {
+            batch.delete_key_prefix(self.context.base_key());
+            self.stored_value = Box::default();
+            self.stored_hash = None;
+        } else if let Some(value) = self.update.take() {
             let key = self.context.base_tag(KeyTag::Value as u8);
             batch.put_key_value(key, &value)?;
             self.stored_value = value;
         }
         let hash = *self.hash.get_mut();
+        // In the scenario where we do a clear
+        // and stored_hash = hash, we need to update the
+        // hash, otherwise, we will recompute it while this
+        // can be avoided.
```

Comment:
remove comment (there is no more trickery)

Reply:
Ok.